### PR TITLE
Change pom files to address maven relocation

### LINKED ISSUE #28059: [FEATURE REQ] Spring on Azure maven relocation
As title,

### PR DESCRIPTION
**Pull request by saragluna**
Fixes #28059.

### CODE REVIEW

File: sdk/boms/azure-spring-boot-bom/pom.xml [hunk 1]
```diff
@@ -2,187 +2,39 @@
 <project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
-
   <groupId>com.azure.spring</groupId>
   <artifactId>azure-spring-boot-bom</artifactId>
-  <version>3.14.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-boot-bom;current} -->
+  <version>3.15.0-beta.1</version>
```
Comment (backwind1233) [inline sdk/boms/azure-spring-boot-bom/pom.xml:7]: will you change the version in aother PR?

Comment (saragluna) [reply sdk/boms/azure-spring-boot-bom/pom.xml]: updated.

**Review by backwind1233 — APPROVED** [review]
LGTM

**Comment by saragluna** [comment]
Closing this PR now, the relocation artifacts have been published.

**Pull request closed without merging**